lib/1.4/dml-builtins.dml

```diff
@@ -1438,39 +1438,37 @@ template bank is (object, shown_desc, _qname) {
     // this carried semantics in DML 1.2; deprecated in 1.4
     param _confidentiality          = undefined;
 
-    session register *_reginfo_cache;
-    session uint64 _reginfo_cache_size = 0;
-
-    shared method _build_register_cache() {
-        if (_reginfo_cache == NULL) {
-            foreach r in (this._each_register) {
-                if (r.offset != unmapped_offset) {
-                    _reginfo_cache_size++;
-                }
+    shared independent startup memoized method _reginfo_cache()
+        -> (register *, uint64) /* cache, cache_size */ {
```

Comment:
`const register *` ?

Reply:
Oh yes.
I like that in our entire design discussion re. data shared across device instances we somehow missed that an independent memoized returning a malloced pointer is essentially a shared variable. Oh well, at least we don't implicitly encourage it.